controller/statsyncer/statsyncer.go

```diff
@@ -0,0 +1,243 @@
+}
+
+func Register(workload *config.WorkloadContext) {
+	s := &StatSyncer{
+		clusterName:  workload.ClusterName,
+		Clusters:     workload.Cluster.Cluster.Clusters(""),
+		ClusterNodes: workload.Cluster.Cluster.ClusterNodes(""),
+		Pods:         workload.K8sClient.CoreV1().Pods(""),
+		Nodes:        workload.K8sClient.CoreV1().Nodes(),
+	}
+
+	go s.syncResources(syncInterval)
+}
+
+func (s *StatSyncer) syncResources(syncInterval time.Duration) {
+	for range time.Tick(syncInterval) {
+		err := s.updateClusterNodeResources()
+		logrus.Info("Syncing allocated resources")
+		if err != nil {
+			logrus.Warn(err)
+		}
+	}
+}
+
+func (s *StatSyncer) updateClusterNodeResources() error {
+	cluster, err := s.getCluster()
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			logrus.Info("Skip syncing node resources, cluster [%s] not found", s.clusterName)
+			return nil
+		}
+		return err
+	}
+	if cluster == nil {
+		logrus.Info("Skip syncing node resources, cluster [%s] deleted", s.clusterName)
+		return nil
+	}
+	if !isProvisioned(cluster) {
+		return fmt.Errorf("Skip syncing node resources - cluster [%s] not provisioned yet", s.clusterName)
+	}
+	nodes, err := s.Nodes.List(metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("Skip syncing node resources - Error getting nodes %v", err)
+	}
+	cnodes, err := s.ClusterNodes.List(metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("Skip syncing node resources - Error getting cluster nodes %v", err)
+	}
+	nodeNameToNode := s.getNodeMapping(nodes)
+	// ans, _ := json.Marshal(nodeNameToNode)
+	// logrus.Infof("nodes mapping %v", string(ans))
+	for _, cnode := range cnodes.Items {
+		node := nodeNameToNode[cnode.NodeName]
+		if node == nil {
+			logrus.Infof("Skip adding cluster node resources [%s] Error getting Node %v", node.Name, err)
+			continue
+		}
+		pods, err := s.getNonTerminatedPods(node.Name)
+		if err != nil {
+			logrus.Infof(err.Error())
+			continue
+		}
+		var nodeData map[string]map[string]map[v1.ResourceName]resource.Quantity
+		if pods != nil {
+			//podName -> req/limit -> data
+			nodeData = make(map[string]map[string]map[v1.ResourceName]resource.Quantity)
+			for _, pod := range pods.Items {
+				nodeData[pod.Name] = make(map[string]map[v1.ResourceName]resource.Quantity)
+				requests, limits := s.getPodData(&pod)
+				nodeData[pod.Name]["requests"] = requests
+				nodeData[pod.Name]["limits"] = limits
+			}
+		}
+		nodeRequests, nodeLimits := s.aggregate(nodeData)
+		err = s.updateClusterNode(&cnode, nodeRequests, nodeLimits)
+		if err != nil {
+			return fmt.Errorf("Error updating cluster node resources [%s] %v", cnode.Name, err)
+		}
+		// ans, _ := json.Marshal(cnode)
+		// logrus.Infof("cnode %v", string(ans))
+	}
+	return nil
+}
+
+func (s *StatSyncer) updateClusterNode(cnode *clusterv1.ClusterNode, requests map[v1.ResourceName]resource.Quantity, limits map[v1.ResourceName]resource.Quantity) error {
+	if cnode.Status.Requested == nil {
```

Comment:
Hm, we should set those values on clusterNode create. And assume that the values are never nil on update